frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -174,7 +174,11 @@ export function InstantSearchProvider({
                .split('/')
                .filter((part) => part !== '');
             const deviceHandle = pathParts.length >= 2 ? pathParts[1] : '';
-            const itemType = pathParts.length >= 3 ? pathParts[2] : '';
+            let itemType = pathParts.length >= 3 ? pathParts[2] : '';
+
+            if (itemType.endsWith('_')) {
+               itemType = itemType.substring(0, itemType.length - 1);
+            }
```

Comment:
I'm not 100% sure what's going on with the itemType here. Does using an itemType in the filterObject cause it to be added to the facets list? I could see Algolia doing that just to show the selected facet even if it's 0, though it is a little odd. Maybe our own code is doing that in the place where we filter facet options? I'm not sure. In any case, I think a cleaner way to accomplish this is to leave the itemType as a const, but down on line 192 when we destylizeDeviceItemType, we can add a .trim() call like `destylizeDeviceItemType(decode...).trim()` will drop any leading or trailing whitespace.

Reply:
Resolved 503da96d26a0f2e1a458a6b4a8ccccdd18913412